cmd/upgrade.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/railwayapp/cli/constants"
+	"github.com/railwayapp/cli/entity"
+)
+
+var OriginalInstallationMethod string
+
+func getExectuablePath() string {
+	ex, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+	exPath := filepath.Dir(ex)
+	return exPath
+}
+
+func getUpdateCommand(installationMethod string) (*exec.Cmd, error) {
+	if installationMethod == "brew" {
+		return exec.Command("brew", "upgrade", "railway"), nil
+	} else if installationMethod == "curl" {
+		return exec.Command("curl", "-fsSL", "https://railway.app/install.sh", "|", "sh"), nil
+	} else if installationMethod == "npm" {
+		currentPath := getExectuablePath()
+		if strings.Contains(currentPath, "npm") {
```

Comment:
(2/2) entirely with this, such that we can do it based on path detection and not have to change the binaries build input (and thus distribute many different binaries)